### DISCUSSION TITLE
SalesforceToGcsOperator RecursionError:maximum recursion depth exceeded while calling a Python object.

### DISCUSSION THREAD

**Discussion by apallerlamudi** [apache/airflow#28703] [Q&A]
Hello All,

I'm using SalesforceToGcsOperator to fetch data from the SF instance and DAG was running fine until Dec 27, 2022 and it started to fail with the below error message from Dec 28, 2022:
RecursionError: maximum recursion depth exceeded while calling a Python object.

Did something change on with respect to this operator : SalesforceToGcsOperator

**Comment by potiuk** [comment]
I thinkk you should rather ask yourself what has changed in your deployment - did you install a new version of airflow/provider? Any changes in your DAGs using them? Did you update container image containing those, etc? 

And only you can answer those questions, I am afraid.

We also have no idea what your deployment is so I can just give you some generic advises.

By definition, when you have python packages installed, they do not change until you change them in your deployment. We cannot remotely make a change in your system - that would be impossible and actually quite serious security issue if we could.

Another possibility (and this is what you should check with the provider of the service - Salesforce or GCS - they could have changed something in their service implementation that triggers this issue - but this is beyond the scope of Airflow and then you have few options:

* upgrade to latest version of Google Provider in the hope that this error has been fixed there (and you can even take a look at the provider's changelog to check it

* check with their release notes of Salesforce/GCS to see if there were any changes and if you find an issue and reproducible way to trigger it and see that it's still not fixed in the latest version of the Google Provider, You can create a provider issue in airlfow issues

* If you are completely lost (after exhausting the options above), I think you can post the stack trace from the latest google provider version and maybe we can help you with investigating of the problem you had. But error message is not nearly enough for that - we need to get stack trace, more information about circumstances it gets triggered, DAG code that triggers them and generally any evidences you can get.

Good luck with your investigations.

**Reply by apallerlamudi** [reply]
Hi,

Thanks for the quick response. yes, we did install a new version of airflow and made a change to the DAG. The issue is now fixed.

Appreciate the help!
